crates/eww/src/config/system_stats.rs

```diff
@@ -153,8 +153,54 @@ pub fn get_battery_capacity() -> Result<String> {
     Ok(json)
 }
 
+#[cfg(any(target_os = "netbsd", target_os = "freebsd", target_os = "openbsd"))]
+pub fn get_battery_capacity() -> Result<String> {
+    let batteries = String::from_utf8(
+        // I have only tested `apm` on FreeBSD, but it *should* work on all of the listed targets,
+        // based on what I can tell from their online man pages.
+        std::process::Command::new("apm")
+            .output()
+            .context("\nError while getting the battery values on bsd, with `apm`: ")?
+            .stdout,
+    )?;
+
+    // `apm` output should look something like this:
+    // $ apm
+    // ...
+    // Remaining battery life: 87%
+    // Remaining battery time: unknown
+    // Number of batteries: 1
+    // Battery 0
+    //         Battery Status: charging
+    //         Remaining battery life: 87%
+    //         Remaining battery time: unknown
+    // ...
+    // last 4 lines are repeated for each battery.
+    // see also:
+    // https://www.freebsd.org/cgi/man.cgi?query=apm&manpath=FreeBSD+13.1-RELEASE+and+Ports
+    // https://man.openbsd.org/amd64/apm.8
+    // https://man.netbsd.org/apm.8
+    let mut json = String::from('{');
+    let re_total = regex!(r"(?m)^Remaining battery life: (\d+)%");
+    let re_single = regex!(r"(?sm)^Battery (\d+):.*?Status: (\w+).*?(\d+)%");
+    for bat in re_single.captures_iter(&batteries) {
+        json.push_str(&format!(
+            r#""BAT{}": {{ "status": "{}", "capacity": {} }}, "#,
+            bat.get(1).unwrap().as_str(),
+            bat.get(2).unwrap().as_str(),
+            bat.get(3).unwrap().as_str(),
```

Comment:
Haven't programmed in months, but I wrote magic vars.
Maybe better to use ? instead of unwrap. Magic variables are a extra, and if they break, they shouldn't take down all of eww.
And it sounds like this is going to break on a lot of bsd distributions (which is fine! Something is better than nothing here)

Reply:
Should be fine in this case, given that regex should guarantee that there will always be three captures,  when the regex pattern has three capture groups, no? Or am I missing something...